src/main/java/cloud/eppo/FlagEvaluator.java

```diff
@@ -0,0 +1,167 @@
+package cloud.eppo;
+
+import static cloud.eppo.ShardUtils.getShard;
+import static cloud.eppo.Utils.base64Decode;
+
+import cloud.eppo.model.ShardRange;
+import cloud.eppo.ufc.dto.Allocation;
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.FlagConfig;
+import cloud.eppo.ufc.dto.Shard;
+import cloud.eppo.ufc.dto.Split;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.Variation;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class FlagEvaluator {
```

Comment:
Mostly not new code; largely adapted from the Android implementation ([source](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/main/java/cloud/eppo/android/FlagEvaluator.java#L20))